src/core/net/dnssd_server.cpp

```diff
@@ -1209,6 +1219,8 @@ void Server::HandleTimer(void)
 
         if (query.GetExpireTime() <= now)
         {
+            mCounters.mHandledByUpstream.mFailedQueries.mNoResponse++;
+            mCounters.mHandledByUpstream.mFailedQueries.mTotalFailure++;
             otPlatDnsCancelUpstreamQuery(&GetInstance(), &query);
         }
```

Comment:
Another code path that can cancelif from `Stop()` where `ResetUpstreamQueryTransaction(txn, kErrorFailed);` is called.

Here is my suggestion for a good way to ensure we update counters on all cases:
- We update counter in `ResetUpstreamQueryTransaction()` where we free an txn and we get `error`
- We need to update the code to make sure `ResetUpstreamQueryTransaction()` is called with porper error in all code paths (e.g. when `aResponseMessage` is `nullptr` we should treat it as `timeout` error.

Reply:
I removed timeout errors, and just use the total failure counter to track now